core/src/main/java/hudson/model/Run.java

```diff
@@ -954,7 +954,10 @@ protected void dropLinks() {
      */  
     public @Nonnull List<RunT> getPreviousBuildsOverThreshold(int numberOfBuilds, @Nonnull Result threshold) {
         RunT r = getPreviousBuild();
-        return r.getBuildsOverThreshold(numberOfBuilds, threshold);
+        if (r != null) {
+            return r.getBuildsOverThreshold(numberOfBuilds, threshold);
+        }
+        return new ArrayList<>(numberOfBuilds);
```

Comment:
NIT-Non-blocking: Probably makes more sense for it to have no initial capacity
```suggestion
        return new ArrayList<>();
```

Reply:
I agree but I chose to give it a capacity as this is what it would have returned in this case before being refactored.